Slim/Player/Protocols/HTTP.pm

```diff
@@ -301,8 +397,18 @@ sub sysread {
 		#$log->debug("Reduced chunksize to $chunkSize for metadata");
 	}
 
-	my $readLength = CORE::sysread($self, $_[1], $chunkSize, length($_[1] || ''));
-
+	my $readLength;
+	
+	# do not read if we are building-up too much processed audio
+	if (${*$self}{'audio_buildup'} > $chunkSize) {
+		${*$self}{'audio_buildup'} = ${*$self}{'audio_process'}->(${*$self}{'audio_stash'}, $_[1], $chunkSize); 
+	} else {	
+		$readLength = CORE::sysread($self, $_[1], $chunkSize, length($_[1] || ''));
```

Comment:
Please properly initialize and name function parameters instead of referring to them using their index. That's fine for short functions where we only reference it once or twice, but not in a function complex like this. Plus if somebody decided to `shift` parameters further up the code you'd fail here.

Reply:
Here I need help with Perl sub passing by reference. If I name the parameter like doing my my ($a, $b) = @_ then I get them by value and I cannot modify them anymore, I becomes a scalar which is a copy. I had that issue and tried to read about it. My understanding is that I needed that @_ direct access (which is the actual reference as Perl pushed references to parameters in @_). See https://stackoverflow.com/questions/3011653/what-is-the-magic-behind-perl-read-function-and-buffer-which-is-not-a-ref. Other than that, I hate to use $_[] for sure!

Reply:
Oh... one more thing I learned today! Do you have to change the original value here?

Reply:
yes, the process function must change the value